app/src/pages/project/SessionsTable.tsx

```diff
@@ -0,0 +1,305 @@
+    },
+    {
+      header: "total tokens",
+      accessorKey: "tokenUsage.total",
+      enableSorting: false,
+      minSize: 80,
+      cell: ({ row, getValue }) => {
+        const value = getValue();
+        if (value === null) {
+          return "--";
+        }
+        const { prompt, completion } = row.original.tokenUsage;
+        return (
+          <TokenCount
+            tokenCountTotal={value as number}
+            tokenCountPrompt={prompt || 0}
+            tokenCountCompletion={completion || 0}
+          />
+        );
+      },
+    },
+    {
+      header: "total traces",
+      accessorKey: "numTraces",
+      enableSorting: false,
+      cell: IntCell,
+    },
+  ];
+  useEffect(() => {
+    //if the sorting changes, we need to reset the pagination
+    // const sort = sorting[0];
+    startTransition(() => {
+      refetch(
+        {
+          after: null,
+          first: PAGE_SIZE,
+        },
+        { fetchPolicy: "store-and-network" }
+      );
+    });
+  }, [sorting, refetch, filterCondition, fetchKey]);
+  const fetchMoreOnBottomReached = React.useCallback(
+    (containerRefElement?: HTMLDivElement | null) => {
+      if (containerRefElement) {
+        const { scrollHeight, scrollTop, clientHeight } = containerRefElement;
+        //once the user has scrolled within 300px of the bottom of the table, fetch more data if there is any
+        if (
+          scrollHeight - scrollTop - clientHeight < 300 &&
+          !isLoadingNext &&
+          hasNext
+        ) {
+          loadNext(PAGE_SIZE);
+        }
+      }
+    },
+    [hasNext, isLoadingNext, loadNext]
+  );
+  const [expanded, setExpanded] = useState<ExpandedState>({});
+  const columnVisibility = useTracingContext((state) => state.columnVisibility);
+  const table = useReactTable<TableRow>({
+    columns,
+    data: tableData,
+    onExpandedChange: setExpanded,
+    manualSorting: true,
+    state: {
+      sorting,
+      expanded,
+      columnVisibility,
+    },
+    enableSubRowSelection: false,
+    onSortingChange: setSorting,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getExpandedRowModel: getExpandedRowModel(),
+    getRowId: (row) => row.id,
+  });
+  const rows = table.getRowModel().rows;
+  const isEmpty = rows.length === 0;
+  return (
+    <div css={spansTableCSS}>
+      <View
+        paddingTop="size-100"
+        paddingBottom="size-100"
+        paddingStart="size-200"
+        paddingEnd="size-200"
+        borderBottomColor="grey-300"
+        borderBottomWidth="thin"
+        flex="none"
+      >
+        <Flex direction="row" gap="size-100" width="100%" alignItems="center">
+          <SpanFilterConditionField onValidCondition={setFilterCondition} />
+        </Flex>
+      </View>
+      <div
+        css={css`
+          flex: 1 1 auto;
+          overflow: auto;
+        `}
+        onScroll={(e) => fetchMoreOnBottomReached(e.target as HTMLDivElement)}
```

Comment:
hmm why does this need to be cast?

Reply:
Oh isee now, hmm i know this is just copied over but this seems a bit weird to me, it seems like the callback should either use the event.target or use the ref directly... it's fine to keep it but could maybe be cleaned up a bit